### PR TITLE
Conditional assignment

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

File: config/enabled.yml [hunk 1]
```diff
@@ -153,6 +153,13 @@ Style/CommentIndentation:
   Description: 'Indentation of comments.'
   Enabled: true
 
+Style/ConditionalAssignment:
+  Description: >-
+                 Use the return value of `if` and `case` statements for
+                 assignment to a variable instead of assigning that variable
+                 inside of each branch
```
Comment (alexdowad) [inline config/enabled.yml]: Small tweak: end sentence with a period.

**Comment by alexdowad** [comment]
:+1:

**Comment by bbatsov** [comment]
31st of July? :-) This has been in the works for quite a while. :-)

**Comment by rrosenblum** [comment]
> 31st of July? :-) This has been in the works for quite a while. :-)

Yeah, I have been working on this on and off for quite a while. The design has completely changed on it at least twice.

**Comment by bbatsov** [comment]
The code is a bit dense (a few comments here and there might be useful), but overall it looks OK to me. I noticed the tests were pretty exhaustive and that's the most important thing. :+1:

**Comment by rrosenblum** [comment]
I will add some comments and submit a new PR.

**Pull request merged**